src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -2304,6 +2307,114 @@ class InsertStatementSegment(BaseSegment):
     )
 
 
+@ansi_dialect.segment()
+class MergeStatementSegment(BaseSegment):
+    """A `MERGE` statement."""
+
+    type = "merge_statement"
+    match_grammar = StartsWith(
+        Ref("MergeIntoLiteralGrammar"),
+    )
+    # Note separate `match_grammar` as overridden in other dialects.
+    parse_grammar = Sequence(
+        Ref("MergeIntoLiteralGrammar"),
+        Indent,
+        OneOf(
+            Ref("TableReferenceSegment"),
+            Ref("AliasedTableReferenceGrammar"),
+        ),
+        Dedent,
+        "USING",
+        Indent,
+        OneOf(
+            Ref("TableReferenceSegment"),
+            Ref("AliasedTableReferenceGrammar"),
+            Sequence(
+                Bracketed(
+                    Ref("SelectableGrammar"),
+                ),
+                Ref("AliasExpressionSegment"),
+            ),
+        ),
+        Dedent,
+        Ref("JoinOnConditionSegment"),
+        Ref("MergeMatchSegment"),
+    )
+
+
+@ansi_dialect.segment()
+class MergeMatchSegment(BaseSegment):
+    """Contains dialect specific merge operations.
+
+    Hookpoint for dialect specific behavior
+    e.g. UpdateClause / DeleteClause, multiple MergeMatchedClauses
+    """
+
+    type = "merge_match"
+    match_grammar = Sequence(
+        Ref("MergeMatchedClauseSegment"),
+        Ref("MergeNotMatchedClauseSegment"),
+    )
+
+
+@ansi_dialect.segment()
+class MergeMatchedClauseSegment(BaseSegment):
+    """The `WHEN MATCHED` clause within a `MERGE` statement."""
+
+    type = "merge_when_matched_clause"
+    match_grammar = Sequence(
+        "WHEN",
+        "MATCHED",
+        Indent,
+        "THEN",
+        Ref("MergeUpdateClauseSegment"),
+        Dedent,
+    )
+
+
+@ansi_dialect.segment()
+class MergeNotMatchedClauseSegment(BaseSegment):
+    """The `WHEN NOT MATCHED` clause within a `MERGE` statement."""
+
+    type = "merge_when_not_matched_clause"
+    match_grammar = Sequence(
+        "WHEN",
+        "NOT",
+        "MATCHED",
+        "THEN",
+        Indent,
+        Ref("MergeInsertClauseSegment"),
+        Dedent,
+    )
+
+
+@ansi_dialect.segment()
+class MergeUpdateClauseSegment(BaseSegment):
+    """`UPDATE` clause within the `MERGE` statement."""
+
+    type = "merge_update_clause"
+    match_grammar = Sequence(
+        "UPDATE",
+        Ref("SetClauseListSegment"),
```

Comment:
Maybe some indents/dedents here?

```suggestion
        Indent,
        Ref("SetClauseListSegment"),
        Dedent,
```